lib/internal/streams/operators.js

```diff
@@ -118,27 +144,26 @@ function map(fn, options) {
             continue;
           }
 
-          if (typeof val?.catch === 'function') {
-            val.catch(onDone);
-          }
+          val = PromiseResolve(val);
+          cnt += 1;
+
 
-          queue.push(val);
+          queue.push(PromisePrototypeThen(val, onThen, onCatch));
           if (next) {
             next();
             next = null;
           }
 
-          if (!done && queue.length && queue.length >= concurrency) {
+          if (!done && cnt >= concurrency) {
             await new Promise((resolve) => {
               resume = resolve;
             });
           }
         }
-        queue.push(kEof);
+        queue.push([kEof, null]);
       } catch (err) {
         const val = PromiseReject(err);
-        PromisePrototypeThen(val, undefined, onDone);
-        queue.push(val);
+        queue.push(PromisePrototypeThen(val, undefined, onCatch));
```

Comment:
Also wrong for the same reason as above.

Reply:
commented